### PR TITLE
Feature/add send to datadog

### PR DESCRIPTION
**Pull request by jonmoss-konsentus**
PR to provide visibility for everyone!

Notable changes:
- Add JSDoc to describe the objects being created.
- Adding the ability to send the build metrics to Datadog, an org-wide secret has been added called DATADOG_TOKEN.
- Allow either sending to Datadog or save the filesystem (both will work together)

Still to come... Unit Tests!!

### CODE REVIEW

**Review by jonmoss-konsentus — COMMENTED** [review]
To see the logs in Datadog please look here 👀 
https://app.datadoghq.com/logs?cols=%40job.total_time_ms%2C%40step.number%2C%40step.total_time_ms&event&from_ts=1591350740442&index=main&live=true&messageDisplay=inline&query=service%3Agithub-actions&stream_sort=desc&to_ts=1591351640442

File: action.yml [hunk 1]
```diff
@@ -1,13 +1,21 @@
-name: 'Collect Github Build Metrcis'
-description: 'Uses github API to collect build metrics'
+name: 'Collect build metrics'
+description: 'Will collect build metrics and return as an array of objects'
```
Comment (andymed-jlp) [inline action.yml:2]: Probably some mention of Datadog would be good as it does that as well

**Review by andymed-jlp — APPROVED** [review]
Seems good to me.

File: src/helpers/index.js [hunk 2]
```diff
@@ -1,14 +1,34 @@
...
 };
 
+/**
+ * Removes any jobs that havent finished.
+ *  Because this action runs at the end of a workflow, this job is never completed, we want to remove this job from the output as to keep our results clean.
+ *
+ * @param {Object[]} jobs The array of unparsed jobs.
+ * @returns {Object[]} An array with only finished jobs.
+ */
 const filterFinishedJobs = jobs => lodash.filter(jobs, { status: 'completed' });
```
Comment (andymed-jlp) [inline src/helpers/index.js:24]: Not sure you need to use lodash, the normal js filter could be used.

File: src/helpers/index.js [hunk 3]
```diff
@@ -1,14 +1,34 @@
...
 
+/**
+ * converts a date to UTC format.
+ *
+ * @param {Date} date The start date
+ * @returns {Date} A UTC datetime with format 'YYYY-M-DD HH:mm:ss.SSS'
+ */
 const convertToUtc = date => moment.utc(date).format('YYYY-M-DD HH:mm:ss.SSS');
```
Comment (andymed-jlp) [inline src/helpers/index.js:33]: Is this format specific or can a standard ISO format be used?

File: src/sendToDatadog/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+const request = require('request');
+const core = require('@actions/core');
```
Comment (andymed-jlp) [inline src/sendToDatadog/index.js:2]: We've use axios for http comms. I presume this is simpler given the use case you're employing and so axios would just make things more complex?

File: src/buildOutput/index.js [hunk 5]
```diff
@@ -1,20 +1,64 @@
...
-    event_type: eventType,
-    jobs: finishedJobs.map(job => parseJob(job)),
-  };
-  return output;
+
+  const result = [];
+
+  finishedJobs.forEach(job => {
+    job['steps'].forEach(step => {
+      let output = {
```
Comment (nickbogaerts) [inline src/buildOutput/index.js:45]: could be a const, or you could just wrap the result.push() around the object

**Review by nickbogaerts — APPROVED** [review]

**Pull request merged**